app/controllers/security_ruleset_controller.rb

```diff
@@ -0,0 +1,41 @@
+
+class SecurityRulesetController < ApplicationController
+  include RulesetValidations
+
+  before_action :validate_file_upload!, only: [:create]
+  before_action :validate_ruleset_name!, only: [:new]
+  before_action :clear_errors
+
+  def new
+  end
+
+  def create
+    crunch_json = JSON.parse(crunch42_data)
+    @score = crunch_json['score']
+    @openapi = crunch_json['openapiState']
+    @counter = crunch_json['issueCounter']
+    @criticality = crunch_json['criticality']
+    @issues = crunch_json['data']['issues'].map do |issue|
```

Comment:
The standard way to iterate over a hash in ruby is to split the iterator into key and value. That makes the intention a lot clearer.

My first impression on seeing this code was that `crunch_json['data']['issues']` was an array. I was then a little confused as to why the second element in the array was so significant.

It was only after examining the test data (a process made a little more complicated by the fact the JSON is stored unprettified) when it then became obvious that `crunch_json['data']['issues']` returns a hash.

If this line is changed to:
```ruby
@issues = crunch_json['data']['issues'].map do |_key, issue|
```
it becomes a lot clearer that we are iteration over a hash and using the values in that hash.

Reply:
For starters we probably want a more intention revealing variabl. crunch_json doesn't make any sense since the json has been parsed.

Reply:
Just noticed now, this is not a hash but an array, so the above format doesn't apply:
![image](https://user-images.githubusercontent.com/46826509/219384147-d6a31027-6d06-4cd6-be03-0c358f33ed77.png)

Reply:
I think the json format you are referring to is the VCR cassette which is a recording of the response from 42 crunch so we can use it in our integration tests.

Reply:
No it's not. Your pry is acting on the output of one of the elements you are iterating over. I presume you added the `binging.pry` after line 18. The `issue` the pry is acting on is an iteration and contains the current key/value pair from the hash.